e2e/testcases/oci_signature_verification_test.go

```diff
@@ -63,7 +63,7 @@ func TestAddPreSyncAnnotationRepoSync(t *testing.T) {
 	nt.Must(rootSyncGitRepo.Add(nomostest.StructuredNSPath(repoSyncID.Namespace, repoSyncID.Name), repoSyncOCI))
 	nt.Must(rootSyncGitRepo.CommitAndPush("Set the RepoSync to sync from OCI"))
 	nt.Must(nt.Watcher.WatchObject(kinds.RepoSyncV1Beta1(), repoSyncID.Name, namespaceRepo,
-		testwatcher.WatchPredicates(testpredicates.RepoSyncHasSourceError(status.SourceErrorCode, "no signatures found"))))
+		testwatcher.WatchPredicates(testpredicates.RepoSyncHasSourceError(status.APIServerErrorCode, "no signatures found"))))
```

Comment:
There's no problem accessing the API server when the webhook validation fails. 
https://cloud.google.com/kubernetes-engine/enterprise/config-sync/docs/reference/errors#knv2002
If we're going with this change, the error reference doc needs to be updated.

Reply:
The way we use APIServerError today, it is also sent for client-side errors, like http timeouts, context cancellations, etc. It's also used for server-side errors like authn/authz, as well as invalid requests, etc.

So realistically, the "APIServerError" name doesn't really describe what the root cause is. But if we want to change that, we would need to do it everywhere (it's used in like 50 places).

But for now, I can update the doc to explain that it's more general and has many possible root causes.